app/search_builders/curation_concerns/filter_suppressed_with_roles.rb

```diff
@@ -0,0 +1,26 @@
+module CurationConcerns
+  # Overrides FilterSuppressed filter to hide documents marked as
+  # suppressed when the current user is permitted to take no workflow
+  # actions for the work's current state
+  #
+  # Assumes presence of `blacklight_params[:id]` and a SolrDocument
+  # corresponding to that `:id` value
+  module FilterSuppressedWithRoles
+    extend ActiveSupport::Concern
+    include CurationConcerns::FilterSuppressed
+
+    def only_active_works(solr_parameters)
+      super if user_work_permissions.empty?
```

Comment:
Can we put this in a method called `current_user_has_active_role?` and then do: `return super unless current_user_has_active_role?`